core/src/main/java/com/spotify/metrics/net/NetstatMetricSet.java

```diff
@@ -0,0 +1,130 @@
+import com.spotify.metrics.core.DerivedLongGauge;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricSet;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class NetstatMetricSet implements SemanticMetricSet {
+
+    private static final String SPACES = "\\s+";
+    private static final String PROC_SNMP_PATH = "/proc/net/snmp";
+    private static final String PROC_NETSTAT_PATH = "/proc/net/netstat";
+
+    private final Supplier<Data> dataSupplier;
+
+    public NetstatMetricSet() {
+        this(Suppliers.memoizeWithExpiration(NetstatMetricSet::readData, 10, TimeUnit.SECONDS));
+    }
+
+    @VisibleForTesting
+    NetstatMetricSet(final Supplier<Data> dataSupplier) {
+        this.dataSupplier = checkNotNull(dataSupplier, "dataSupplier");
+    }
+
+    @Override
+    public Map<MetricId, Metric> getMetrics() {
+        final Map<MetricId, Metric> gauges = new HashMap<>();
+        gauges.put(
+            MetricId.build().tagged("what", "segments-retransmitted", "unit", "tcp-segments/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().segmentsRetransmitted;
+                }
+            });
+        gauges.put(
+            MetricId.build().tagged("what", "bad-segments-received", "unit", "tcp-segments/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().badSegmentsReceived;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-timeouts", "unit", "timeouts/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpTimeouts;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-lost-retransmit", "unit", "retransmit/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpLostRetransmit;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-syn-retrans", "unit", "retransmit/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpSynRetrans;
+                }
+            });
+
+        return gauges;
+    }
+
+    private static Data readData() {
+        return readData(PROC_NETSTAT_PATH, PROC_SNMP_PATH);
+    }
+
+    @VisibleForTesting
+    static Data readData(final String netstatPath, final String snmpPath) {
+        final Data data = new Data();
+        final Map<String, Long> netstatValues = readFile(netstatPath, "TcpExt:");
```

Comment:
Would it make sense to do the check of if we're on a L*nix system before initializing, and just deactivate if the `/proc/sys/...` don't exist?

Reply:
I think that the end result will be the same and having that logic might be more complex than just fail during parsing. It's a good point about this only working on *nix system. Will add that to the docs